src/NexusMods.App.UI/Pages/Downloads/InProgressView.axaml.cs

```diff
@@ -108,26 +126,60 @@ public InProgressView()
                     }
                 })
                 .DisposeWith(d);
+            
+            // Dynamically Update Completed Download Count
+            this.WhenAnyValue(view => view.ViewModel!.CompletedDownloadCount)
+                .OnUI()
```

Comment:
`OnUI` statements should be in the VM, the View shouldn't have to do thread switching.